src/seedu/addressbook/Main.java

```diff
@@ -106,11 +106,24 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        CommandResult result = null;
         try {
             command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+            result = command.execute();
+        } catch (Exception e) {
+            ui.showToUser(e.getMessage());
+            throw new RuntimeException(e);
+        }
+        try{
             storage.save(addressBook);
             return result;
+        } catch (StorageFile.ReadOnlyFileException e) {
+            if (result.getRelevantPersons().isPresent()) {
+                return new CommandResult(
+                        result.feedbackToUser + e.getMessage(), result.getRelevantPersons().get());
+            } else {
+                return new CommandResult(result.feedbackToUser + e.getMessage());
```

Comment:
In general, just showing the error message or printing the stack trace is not considered "good exception handling".
Think how you can improve the user experience in light of the exception that just occurred.